OpenRA.Mods.Common/Traits/CarryableHarvester.cs

```diff
@@ -37,9 +37,8 @@ void INotifyHarvesterAction.MovingToResources(Actor self, CPos targetCell)
 		void INotifyHarvesterAction.MovingToRefinery(Actor self, Actor refineryActor)
 		{
 			var iao = refineryActor.Trait<IAcceptResources>();
-			var location = refineryActor.Location + iao.DeliveryOffset;
 			foreach (var t in transports)
-				t.RequestTransport(self, location);
+				t.RequestTransport(self, self.World.Map.CellContaining(iao.DeliveryPosition));
```

Comment:
Bonus: Move `self.World.Map.CellContaining(iao.DeliveryPosition)` into a variable to avoid needing to make the calculation again during each iteration.